osbs/build/build_requestv2.py

```diff
@@ -174,10 +186,12 @@ def set_required_secrets(self):
             data = reactor_config_override
             required_secrets = data.get(req_secrets_key, [])
             token_secrets = data.get(token_secrets_key, [])
+            self._set_data_from_reactor_config(data)
         elif reactor_config_map:
             config_map = self.osbs_api.get_config_map(reactor_config_map)
             required_secrets = config_map.get_data_by_key('config.yaml').get(req_secrets_key, [])
             token_secrets = config_map.get_data_by_key('config.yaml').get(token_secrets_key, [])
+            self._set_data_from_reactor_config(config_map.get_data_by_key('config.yaml'))
```

Comment:
maybe refactor this to
```
data = {}
if reactor_config_override:
    data = reactor_config_override
elif reactor_config_map:
    data = self.osbs_api.get_config_map(reactor_config_map).get_data_by_key('config.yaml')
required_secrets = data.get(req_secrets_key, [])
token_secrets = data.get(token_secrets_key, [])
self._set_data_from_reactor_config(data)
```
or better yet, make set_data_from_reactor_config(self) the top level call that figures out data, sets the source stuff, and calls _set_required_secrets(data)

Reply:
yeah I'll refactor that